backend/src/routes/appointment.ts

```diff
@@ -0,0 +1,43 @@
+import express from "express";
+import appointmentService from '../services/appointment-service';
+import { sendBackErrorMessage } from "src/routes/errorProcess";
+
+const router = express.Router();
+
+router.post('/', (req, res) => {
+  appointmentService.createItem(req.body)
+    .then(newAppointment => res.status(200).send(newAppointment))
+    .catch(error => sendBackErrorMessage(req, res, error));
+});
+
+router.get('/', (req, res) => {
+  appointmentService.getAllValidItems()
+    .then(allAppointments => res.status(200).send(allAppointments))
+    .catch(error => sendBackErrorMessage(req, res, error));
+});
+
+router.get('/filter', (req, res) => {
+  appointmentService.getItemsByFilter(req.query)
+    .then(filteredAppointments => res.status(200).send(filteredAppointments))
+    .catch(error => sendBackErrorMessage(req, res, error));
+});
+
+router.get('/:id', (req, res) => {
+  appointmentService.getItemById(req.params.id)
+    .then(appointmentItem => res.status(200).send(appointmentItem))
+    .catch(error => sendBackErrorMessage(req, res, error));
+});
+
+router.put('/', (req, res) => {
```

Comment:
I'm suggesting to change the path for `'/:id'`. It looks like we are updating one appointment only, and this is not consistent with RESTful style and may lead to confusion.

Reply:
in the service, id will be extracted and used to update the appointment, so we do not need to send the ID

Reply:
we embed id in the update info body, this is because the sequelize will detect the id inside the body automatically, so I think it is not necesssary to separate the id in the frontend and merge it into the body again in the backend and simply use `/` as the path